File: moviematch_back/src/main/java/com/brademobackend/moviematch/model/Publicidade.java
package com.brademobackend.moviematch.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Publicidade {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id_publicidade;
    private String descricao;
    private String link_imagem;
}

File: projeto_brademo/lib/model/publicidade.dart
class Publicidade {
  final int idPublicidade;
  final String descricao;
  final String linkImagem;

  Publicidade({
    required this.idPublicidade,
    required this.descricao,
    required this.linkImagem,
  });

  factory Publicidade.fromJson(Map<String, dynamic> json) {
    return Publicidade(
      idPublicidade: json['id_publicidade'],
      descricao: json['descricao'],
      linkImagem: json['link_imagem'],
    );
  }
}